src/lib/utils.ts

```diff
@@ -282,23 +283,44 @@ export function encodeKey(
     case 'mappingwithgrouping':
     case 'singleton':
     case 'mapping':
+      if (schema.valueType.includes('[CompactBytesArray]')) {
+        const valueType = schema.valueType.replace('[CompactBytesArray]', '');
+        const valueContent = schema.valueContent.replace(
+          '[CompactBytesArray]',
+          '',
+        );
+
+        if (
+          Array.isArray(value[0]) &&
```

Comment:
By doing `value[0]` you assume that value is an array, but it might not be, as TS shows:
![image](https://user-images.githubusercontent.com/477945/214849670-ec329c1d-35c4-4cb0-a9c9-23298bc46745.png)

So what happens if value is a `number` ? Will it break when you try to call

By adding a first check on the value itself:
![image](https://user-images.githubusercontent.com/477945/214849884-9314633d-6143-4cc0-bae8-7644872e46a2.png)

You make TS smarter, as you can see:
![image](https://user-images.githubusercontent.com/477945/214849947-d467e17f-d8af-4ca1-bfed-48ad54b44990.png)


Now the set of types that can apply to value is restricted to array only.

It also solves __part__ of the type casting below.

Please review this code to make it more "naturally" typescript ready by avoiding using **as** as an "easy way to work around the TS error"